pkg/extend/extend.go

```diff
@@ -83,10 +84,24 @@ func extend(d, fsType string) error {
 		return fmt.Errorf("Partition 1 on disk %s is not a Linux Partition", d)
 	}
 
-	if partition.Start+partition.Size == f.PartitionTable.LastLBA {
+	if f.PartitionTable.Label == "gpt" && partition.Start+partition.Size == f.PartitionTable.LastLBA {
 		log.Printf("No free space on device to extend partition")
 		return nil
 	}
+	if f.PartitionTable.Label == "dos" {
+		out, err := exec.Command("blockdev", "--getsz", d).Output()
+		if err != nil {
+			return fmt.Errorf("Unable to get total size of %s: %v", d, err)
+		}
+		totalSize, err := strconv.Atoi(strings.TrimSpace(string(out)))
+		if err != nil {
+			return fmt.Errorf("Unable to convert total size from string to int: %v", err)
+		}
+		if totalSize == partition.Size+2048 {
```

Comment:
2048 is also partition.Start. I can refactor to have a function that returns the total size of the disk (gpt and dos) and have a single check.

Reply:
Why the magic number vs. using `partition.Start` in this case?

Reply:
Yes it's a mistake, I will fix it.